app/src/main/java/com/android/unio/ui/event/overlay/EventTypeOverlay.kt

```diff
@@ -0,0 +1,156 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import com.android.unio.R
+import com.android.unio.model.event.EventType
+import com.android.unio.model.strings.test_tags.event.EventTypeOverlayTestTags
+import com.android.unio.ui.theme.AppTypography
+
+@Composable
+fun EventTypeOverlay(
+    onDismiss: () -> Unit,
+    onSave: (List<Pair<EventType, MutableState<Boolean>>>) -> Unit,
+    types: List<Pair<EventType, MutableState<Boolean>>>
+) {
+  val context = LocalContext.current
+  val scrollState = rememberScrollState()
+  val copiedTypes = types.toList().map { it.first to mutableStateOf(it.second.value) }
+  val isError = remember { mutableStateOf(false) }
+
+  Dialog(
+      onDismissRequest = onDismiss,
+      properties = DialogProperties(usePlatformDefaultWidth = false)) {
+        Card(
+            elevation = CardDefaults.cardElevation(8.dp),
+            shape = RoundedCornerShape(16.dp),
+            modifier =
+                Modifier.fillMaxWidth().padding(20.dp).testTag(EventTypeOverlayTestTags.CARD)) {
+              Column(
+                  modifier =
+                      Modifier.fillMaxWidth()
+                          .padding(15.dp)
+                          .sizeIn(maxHeight = 400.dp)
+                          .testTag(EventTypeOverlayTestTags.COLUMN),
+                  verticalArrangement = Arrangement.SpaceBetween) {
+
+                    // Text fields for the title and description of the Interest Overlay
+                    Text(
+                        text = context.getString(R.string.event_type_overlay_title),
+                        style = AppTypography.headlineSmall,
+                        modifier = Modifier.testTag(EventTypeOverlayTestTags.TITLE_TEXT))
+                    Text(
+                        text = context.getString(R.string.event_type_overlay_description),
+                        style = AppTypography.bodyMedium,
+                        modifier =
+                            Modifier.padding(bottom = 5.dp)
+                                .testTag(EventTypeOverlayTestTags.SUBTITLE_TEXT))
+                    Surface(
+                        modifier = Modifier.sizeIn(maxHeight = 250.dp), color = Color.Transparent) {
+                          Column(modifier = Modifier.verticalScroll(scrollState)) {
+                            copiedTypes.forEachIndexed { index, pair ->
+
+                              // The row for each interest
+                              EventTypeOverlayRow(
+                                  pair,
+                                  onChange = {
+                                    isError.value = copiedTypes.count({ it.second.value }) > 3
```

Comment:
Consider extracting the value `3` into a constant to make the code more understandable and maintainable.

Reply:
never